### PR TITLE
StandardJS

### PR DESCRIPTION
**Pull request by Schultzer**

### CODE REVIEW

File: config/config.js [hunk 1]
```diff
@@ -1,11 +1,11 @@
 module.exports = exports = function (dbname) {
-  var config = require('./config.json');
-  var node_env = process.env.NODE_ENV || 'production';
-  config.env = node_env;
-  config.smtpTransporter = require('./transporter');
-  config.db = require('./db')(dbname || config.dbname, 'localhost');
+  var config = require('./config.json')
+  var nodeEnv = process.env.NODE_ENV || 'production'
```
Comment (danschultzer) [inline config/config.js]: Just move this down to the next line instead

File: controllers/api_breakpad.js [hunk 2]
```diff
@@ -22,29 +22,31 @@ module.exports = function (authenticate) {
...
-        saveReport(req, file, cb);
-      });
-
-      writeStream.on("error", function(error) {
-        cb(error);
-      });
-  });
+  router.post('/crash-report', upload.single('upload_file_minidump'), function (req, res, next) {
+    var id = uuid.v4()
+    var writeStream
```
Comment (danschultzer) [inline controllers/api_breakpad.js]: Why not just do `var writeStream = gfs. ...`?

File: controllers/api_breakpad.js [hunk 3]
```diff
@@ -22,29 +22,31 @@ module.exports = function (authenticate) {
...
+    writeStream = gfs.createWriteStream({
+      filename: id
+    })
+    var cb = function (error) {
+      if (error) {
```
Comment (danschultzer) [inline controllers/api_breakpad.js]: Probably nicer as a one-liner?

Comment (danschultzer) [reply controllers/api_breakpad.js]: (This also is relevant to all subsequent identical code)

File: controllers/api_receipt.js [hunk 4]
```diff
@@ -1,29 +1,29 @@
...
-    scanner = require('receipt-scanner');
+  var express = require('express')
+  var router = express.Router()
+  var multer = require('multer')
+  var crypto = require('crypto')
+  var mime = require('mime')
+  var scanner = require('receipt-scanner')
 
-  var storage = multer.diskStorage({
+  var storage
```
Comment (danschultzer) [inline controllers/api_receipt.js]: Why not just do `var storage = ...`?

File: controllers/api_receipt.js [hunk 5]
```diff
@@ -35,22 +35,23 @@ module.exports = function () {
...
    */
-  router.post('/receipt', upload.single('document'), function(req, res, next) {
+  router.post('/receipt', upload.single('document'), function (req, res, next) {
     if (!req.file) {
-      throw new Error('Missing file attachment');
+      throw new Error('Missing file attachment')
     }
 
-    console.log(req.file.path);
+    console.log(req.file.path)
```
Comment (danschultzer) [inline controllers/api_receipt.js]: Should be removed, it's debug code.

File: controllers/auth_github.js [hunk 6]
```diff
@@ -1,124 +1,133 @@
...
+  var Strategy = require('passport-github2')
+  var Admin = require('../models/admin')
+  var request = require('request')
+
+  router.use(passport.initialize())
+  passport.serializeUser(function (admin, done) {
+    done(null, admin._id)
+  })
+  passport.deserializeUser(function (id, done) {
+    var User
```
Comment (danschultzer) [inline controllers/auth_github.js]: This doesn't make any sense?

File: models/admin.js [hunk 7]
```diff
@@ -1,28 +1,29 @@
...
+  var currentDate = new Date()
 
   // change the updated_at field to current date
-  this.updated_at = currentDate;
+  this.updated_at = currentDate
 
   // if created_at doesn't exist, add to that field
-  if (!this.created_at)
-    this.created_at = currentDate;
+  if (!this.created_at) {
```
Comment (danschultzer) [inline models/admin.js]: Maybe one-liner?

File: models/bug_report.js [hunk 8]
```diff
@@ -1,40 +1,40 @@
...
+  var currentDate = new Date()
 
   // change the updated_at field to current date
-  this.updated_at = currentDate;
+  this.updated_at = currentDate
 
   // if created_at doesn't exist, add to that field
-  if (!this.created_at)
-    this.created_at = currentDate;
+  if (!this.created_at) {
```
Comment (danschultzer) [inline models/bug_report.js]: Maybe one-liner?

File: models/crash_report.js [hunk 9]
```diff
@@ -1,37 +1,36 @@
...
-  this.updated_at = currentDate;
+  this.updated_at = currentDate
 
   // if created_at doesn't exist, add to that field
-  if (!this.created_at)
-    this.created_at = currentDate;
-
-  next();
-});
+  if (!this.created_at) {
```
Comment (danschultzer) [inline models/crash_report.js]: Maybe one-liner?

File: public/javascripts/receipt.js [hunk 10]
```diff
@@ -1,140 +1,148 @@
...
 
         // If we're there or can't get any closer because of the viewport's
         // size, stop!
-        if ((window.pageYOffset >= destination_y) || (window.pageYOffset === prior_position_y)) {
-            clearInterval(SmoothScrolling.last_event_interval);
-            location.hash = destination_hash;
-        }
+    if ((window.pageYOffset >= destinationY) || (window.pageYOffset === priorPositionY)) {
+      clearInterval(SmoothScrolling.last_event_interval)
+      var location
```
Comment (danschultzer) [inline public/javascripts/receipt.js]: Why this?

**Review by danschultzer — CHANGES_REQUESTED** [review]

File: models/admin.js (hunk 7 above)
Comment (Schultzer) [reply models/admin.js]: Then we not following StandardJS

File: models/bug_report.js (hunk 8 above)
Comment (Schultzer) [reply models/bug_report.js]: Then we not following StandardJS

File: models/crash_report.js (hunk 9 above)
Comment (Schultzer) [reply models/crash_report.js]: Then we not following StandardJS

File: controllers/api_breakpad.js (hunk 2 above)
Comment (Schultzer) [reply controllers/api_breakpad.js]: Then we not following StandardJS

File: controllers/api_breakpad.js (hunk 3 above)
Comment (Schultzer) [reply controllers/api_breakpad.js]: Then we not following StandardJS

File: controllers/api_receipt.js (hunk 4 above)
Comment (Schultzer) [reply controllers/api_receipt.js]: Then we not following StandardJS

File: controllers/api_receipt.js (hunk 5 above)
Comment (Schultzer) [reply controllers/api_receipt.js]: Then we not following StandardJS

File: controllers/auth_github.js (hunk 6 above)
Comment (Schultzer) [reply controllers/auth_github.js]: Variable User is not defined

File: public/javascripts/receipt.js (hunk 10 above)
Comment (Schultzer) [reply public/javascripts/receipt.js]: Variable location is not defined

File: middlewares/api_auth.js [hunk 11]
```diff
@@ -6,9 +6,11 @@ module.exports = function () {
   passport.use(new Strategy(
     function (token, cb) {
       Admin.findOne({ access_token: token }, function (error, admin) {
-        if (error) { return cb(error) }
+        if (error) return cb(error)
 
-        if (!admin) { return cb(null, false) }
+        if (!admin) {
```
Comment (danschultzer) [inline middlewares/api_auth.js]: This could probably be one-liner too.

**Review by danschultzer — CHANGES_REQUESTED** [review]

File: controllers/auth_github.js [hunk 12]
```diff
@@ -1,124 +1,120 @@
...
+  var router = express.Router()
+  var passport = require('passport')
+  var Strategy = require('passport-github2')
+  var Admin = require('../models/admin')
+  var request = require('request')
+
+  router.use(passport.initialize())
+  passport.serializeUser(function (admin, done) { done(null, admin._id) })
+  passport.deserializeUser(function (id, done) {
+    Admin.findById(id, function (err, admin) { done(err, admin) })
```
Comment (danschultzer) [inline controllers/auth_github.js]: The function should be in multiple lines.

File: controllers/auth_github.js [hunk 13]
```diff
@@ -1,124 +1,120 @@
...
-            var options =  {
+  var express = require('express')
+  var router = express.Router()
+  var passport = require('passport')
+  var Strategy = require('passport-github2')
+  var Admin = require('../models/admin')
+  var request = require('request')
+
+  router.use(passport.initialize())
+  passport.serializeUser(function (admin, done) { done(null, admin._id) })
```
Comment (danschultzer) [inline controllers/auth_github.js]: The function should be in multiple lines.

File: helpers/send_report_to_admins.js [hunk 14]
```diff
@@ -8,21 +8,14 @@ module.exports = function (type) {
...
     Admin.find(function (error, admins) {
-      if (error) {
-        return reject(error)
-      }
+      if (error) return reject(error)
 
-      emails = admins.map(function (admin) {
-        return admin.email
-      })
+      emails = admins.map(function (admin) { return admin.email })
```
Comment (danschultzer) [inline helpers/send_report_to_admins.js]: Multiline

File: public/javascripts/receipt.js [hunk 15]
```diff
@@ -1,140 +1,142 @@
...
                 if (json.success) {
-                  file.result.parsed = json.results;
+                  file.result.parsed = json.results
                 } else {
-                  file.result.error = json.error;
+                  file.result.error = json.error
                 }
               } catch (error) {
-                alert(error.message);
+                var alert = alert(error.message)
```
Comment (danschultzer) [inline public/javascripts/receipt.js]: Why is this a var?

**Review by danschultzer — CHANGES_REQUESTED** [review]

**Review by danschultzer — APPROVED** [review]

**Pull request merged**